portalnet/src/overlay_service.rs

```diff
@@ -2402,7 +2402,34 @@ where
             peer_timeout: self.query_peer_timeout,
         };
 
-        let closest_enrs = self.closest_connected_nodes(&target_key, query_config.num_results);
+        let mut closest_enrs = self.closest_connected_nodes(&target_key, query_config.num_results);
+        if closest_enrs.is_empty() {
+            // If there are no connected nodes, use the closest nodes regardless of connectivity.
```

Comment:
This is an concerning enough situation that it seems worth a `warn!` log (or maybe at least `info!`). Having 0 known connected nodes is noteworthy.